Sources/SwiftGen/Logs.swift

```diff
@@ -39,17 +39,37 @@ enum ANSIColor: UInt8, CustomStringConvertible {
   }
 }
 
+// Based on https://github.com/realm/SwiftLint/blob/0.39.2/Source/SwiftLintFramework/Extensions/QueuedPrint.swift
+
 func logMessage(_ level: LogLevel, _ string: CustomStringConvertible) {
-  switch level {
-  case .info:
-    fputs(ANSIColor.green.format("\(string)\n"), stdout)
-  case .warning:
-    fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
-  case .error:
-    fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
+  outputQueue.async {
+    switch level {
+    case .info:
+      fputs(ANSIColor.green.format("\(string)\n"), stdout)
+    case .warning:
+      fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
+    case .error:
+      fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
```

Comment:
I know it's not directly related to the immediate scope of this PR, but could be the occasion to quick-fix https://github.com/SwiftGen/SwiftGen/issues/547 here 🤔

Reply:
i'd rather leave that for another PR since I don't have any context about that issue

Reply:
Agreed, no need to mix that in here.